examples/pong_tutorial_06/main.rs

```diff
@@ -0,0 +1,172 @@
+};
+
+fn main() -> amethyst::Result<()> {
+    amethyst::start_logger(Default::default());
+
+    let app_root = application_root_dir()?;
+    let display_config_path = app_root.join("examples/pong_tutorial_06/config/display.ron");
+
+    let game_data = GameDataBuilder::default()
+        // Add the transform bundle which handles tracking entity positions
+        .with_bundle(TransformBundle::new())?
+        .with_bundle(
+            InputBundle::<StringBindings>::new().with_bindings_from_file(
+                app_root.join("examples/pong_tutorial_06/config/bindings.ron"),
+            )?,
+        )?
+        .with_bundle(UiBundle::<StringBindings>::new())?
+        .with(
+            UiGlyphsSystem::<DefaultBackend>::new(),
+            "ui_glyphs_system",
+            // These dependencies are required because this `System` depends from the `UiBundle` `System`s
+            &[
+                "ui_loader",
+                "ui_transform",
+                "font_processor",
+                "ui_text_editing_mouse_system",
+                "ui_text_editing_input_system",
+            ],
+        )
+        // We have now added our own systems, defined in the systems module
+        .with(systems::PaddleSystem, "paddle_system", &["input_system"])
+        .with(systems::MoveBallsSystem, "ball_system", &[])
+        .with(
+            systems::BounceSystem,
+            "collision_system",
+            &["paddle_system", "ball_system"],
+        )
+        .with(systems::WinnerSystem, "winner_system", &["ball_system"])
+        .with(
+            Processor::<SpriteSheet>::new(),
+            "sprite_sheet_processor",
+            &[],
+        )
+        .with(
+            SpriteVisibilitySortingSystem::new(),
+            "sprite_visibility_sorting_system",
+            &[],
+        )
+        .with_bundle(WindowBundle::from_config_path(display_config_path))?
+        // The renderer must be executed on the same thread consecutively, so we initialize it as thread_local
+        // which will always execute on the main thread.
+        .with_bundle(RenderingBundle::<DefaultBackend, _>::new(
+            ExampleGraph::default(),
+        ))?;
+
+    // This line is not mentioned in the pong tutorial as it is specific to the context
+    // of the git repository. It only is a different location to load the assets from.
+    let assets_dir = app_root.join("examples/assets/");
+
+    let mut game = Application::new(assets_dir, Pong::default(), game_data)?;
+    game.run();
+    Ok(())
+}
+
+// This graph structure is used for creating a proper `RenderGraph` for rendering.
+// A renderGraph can be thought of as the stages during a render pass. In our case,
+// we are only executing one subpass (DrawFlat2D, or the sprite pass). This graph
+// also needs to be rebuilt whenever the window is resized, so the boilerplate code
+// for that operation is also here.
+#[derive(Default)]
+struct ExampleGraph {
```

Comment:
In my opinion, it would have been better to have this in a minimal example (of its own) based on the `renderable` example. This way, it would be easier to find the graph code.

Reply:
Also, usually we reserve pong_tutorial_x entries for when we have a corresponding book tutorial page.